internal/cmd/operator_list_custom_resources.go

```diff
@@ -0,0 +1,71 @@
+package cmd
+
+import (
+	"fmt"
+
+	"github.com/operator-framework/kubectl-operator/internal/cmd/internal/log"
+	"github.com/operator-framework/kubectl-operator/internal/pkg/action"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+)
+
+func newOperatorListCustomResourcesCmd(cfg *action.Configuration) *cobra.Command {
+	CRLister := action.NewOperatorListCustomResources(cfg)
+	cmd := &cobra.Command{
+		Use:     "list-custom-resources",
+		Aliases: []string{"list-crs"},
+		Short:   "List all custom resources on-cluster associated with a particular operator",
+		Args:    cobra.MaximumNArgs(1),
+		Run: func(cmd *cobra.Command, args []string) {
+			if len(args) == 0 {
+				log.Fatal("did not provide operator or package name")
+			}
+			CRLister.PackageName = args[0]
+
+			op, err := CRLister.FindOperator(cmd.Context())
+			if err != nil {
+				log.Fatal(err)
+			}
+			fmt.Printf("found operator %s", op.Name)
+
+			crds, err := CRLister.Unzip(cmd.Context(), &op)
+			if err != nil {
+				log.Fatal(err)
+			}
+			fmt.Printf("found owned crds #%v", crds)
+
+			list, err := CRLister.List(cmd.Context(), crds)
+			if err != nil {
+				log.Fatal(err)
+			}
+
+			cont := "n"
+			i := 0
+		OUTPUT:
+			for {
+				// TODO make this terminate
+				for ; i < i+10; i++ {
+					fmt.Printf("#%v", list.Items[i])
+
+					if !CRLister.ListAll {
+						fmt.Print("Print 10 more custom resources? Enter y to continue listing")
+						_, err := fmt.Scanf("%s", &cont)
+						if err != nil {
+							log.Fatal(err)
+						}
+						if cont != "y" {
+							break OUTPUT
+						}
+					}
+				}
+			}
+		},
+	}
+	bindOperatorListCustomResourcesFlags(cmd.Flags(), CRLister)
+	return cmd
+}
+
+func bindOperatorListCustomResourcesFlags(fs *pflag.FlagSet, CRLister *action.OperatorListCustomResources) {
+	fs.BoolVarP(&CRLister.ListAll, "list-all", "y", true, "List all CRs in the result without batching. Defaults to true.")
+	fs.StringVarP(&CRLister.Namespace, "namespace", "n", "default", "Set namespace in which to look for operator.")
```

Comment:
The `--namespace` flag is configured globally and setup as part of the `config`, so no need to add it to the command.

I _would_ recommend adding the `-A, --all-namespaces` flag to this subcommand (see `kubectl operator list`'s implemetation for an example) so that command line users can optionally view CRs from all namespaces rather than just the namespace setup in their kubeconfig context.

Reply:
My thinking with the `--namespace` flag was that if I am a user in one namespace, I can query for the operator object in another namespace and find the CRs associated with it. I will remove this flag, but this limits the command to only CRs for operators in the current namespace context and across the entire cluster. I think this sounds fine.

Reply:
You misunderstood my command. You'll get `--namespace` for free. You can access it via `config.Namespace`. If `--namespace` is unset, it will be the value of the namespace from the context. If `--namespace` is set, it will be whatever it is set to.

After having talked about this more today, I think we explicitly _don't_ want an `--all-namespaces` flag, because we're always looking for a specific operator in a specific namespace.